### PR TITLE
[Buildbot] Add scripts for build steps

### PR DESCRIPTION
**Pull request by ikurtchen**
The scripts will be called by Buildbot on corresponding steps.

Signed-off-by: Kurt Chen <kurt.chen@intel.com>

### CODE REVIEW

File: buildbot/check-clang.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        b) BRANCH=$OPTARG ;;
+        n) BUILD_NUMBER=$OPTARG ;;
+        r) PR_NUMBER=$OPTARG ;;
+    esac
+done && shift $(($OPTIND - 1))
+
+# we're in llvm.obj dir
+BUILD_DIR=${PWD}
+
+make check-clang VERBOSE=1 LIT_ARGS="-v -j `nproc`"
```
Comment (bader) [inline buildbot/check-clang.sh]: Is it possible to make this utility configurable?
I use ninja to build LLVM and run the tests and I'd like to use these scripts on my local machine to reproduce the results/debug/etc.
I think you can re-use cmake for this: `cmake --build <> --taget <>`

File: buildbot/check-sycl.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        b) BRANCH=$OPTARG ;;
+        n) BUILD_NUMBER=$OPTARG ;;
+        r) PR_NUMBER=$OPTARG ;;
+    esac
+done && shift $(($OPTIND - 1))
+
+# we're in llvm.obj dir
+BUILD_DIR=${PWD}
+
+make check-sycl VERBOSE=1 LIT_ARGS="-v -j `nproc`"
```
Comment (bader) [inline buildbot/check-sycl.sh]: There is no difference between `check-sycl`, `check-llvm`, `check-llvm-spirv` and `check-clang` scripts between except the target name. I think it should be one script with configurable target.

File: buildbot/compile.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+done && shift $(($OPTIND - 1))
+
+# we're in llvm.obj dir
+BUILD_DIR=${PWD}
+
+if [ "$PROJECT" == "sycl" ]; then
+    make -j`nproc`
+elif [ "$PROJECT" == "llorg" ]; then
+    make -j`nproc`
+fi
```
Comment (bader) [inline buildbot/compile.sh]: What is the point of this `if`?
What is the purpose of PROJECT variable?

**Comment by bb-sycl** [comment]
> Is it possible to write cross-platform scripts? We are starting working on Windows support and I have concerns that these might not be suitable for Windows.
> 
> Could you also add README.md file with instructions how to use these scripts, please?

Yes, the script is not suitable for Windows. For Windows, new builders will be created on Buildbot, and *.bat files could be added for new builders.  

I think the purpose of these scripts is for developers to do customize the build command for the builders on Buildbot. So I think it's reasonable to seperate Windows scripts with these scripts, the command may be different and seperate script could keep things simple.

OK, I will add README for it.

File: buildbot/check-clang.sh (hunk 1 above)
Comment (bb-sycl) [reply buildbot/check-clang.sh]: I think the scripts are for builders on Buildbot, Buildbot will run these scripts during build. As per @vladimirlaz , developer will control these scripts, you can change to ninja as you wish, but you need to change all of the scripts, e.g. conigure.sh, compile.sh

**Comment by ikurtchen** [comment]
> Is it possible to write cross-platform scripts? We are starting working on Windows support and I have concerns that these might not be suitable for Windows.
> 
> Could you also add README.md file with instructions how to use these scripts, please?

Yes, the script is not suitable for Windows. For Windows, new builders will be created on Buildbot, and *.bat files could be added for new builders.

I think the purpose of these scripts is for developers to do customize the build command for the builders on Buildbot. So I think it's reasonable to seperate Windows scripts with these scripts, the command may be different and seperate script could keep things simple.

OK, I will add README for it.

File: buildbot/check-clang.sh (hunk 1 above)
Comment (ikurtchen) [reply buildbot/check-clang.sh]: I think the scripts are for builders on Buildbot, Buildbot will run these scripts during build. As per @vladimirlaz , developer will control these scripts, you can change to ninja as you wish, but you need to change all of the scripts, e.g. conigure.sh, compile.sh

File: buildbot/compile.sh (hunk 3 above)
Comment (ikurtchen) [reply buildbot/compile.sh]: OK, we can remove it.

File: buildbot/check-sycl.sh (hunk 2 above)
Comment (ikurtchen) [reply buildbot/check-sycl.sh]: @vladimirlaz could you please help comment on this. I think one of the advantage of seperate script is easy customization. Anyway, we can combine them if needed.

Comment (vladimirlaz) [reply buildbot/check-sycl.sh]: Good catch from my perspective. We can have check.sh script which accepts exact target as command line and environment variable

File: buildbot/check-clang.sh (hunk 1 above)
Comment (vladimirlaz) [reply buildbot/check-clang.sh]: Do we need these complication? If you want to reproduce failure it is better to have exact the same configuration including tooling.
Locally developer can use any tools chain he needs.

**Comment by vladimirlaz** [comment]
> Is it possible to write cross-platform scripts? We are starting working on Windows support and I have concerns that these might not be suitable for Windows.
> 
> Could you also add README.md file with instructions how to use these scripts, please?

**Comment by vladimirlaz** [comment]
> Is it possible to write cross-platform scripts? We are starting working on Windows support and I have concerns that these might not be suitable for Windows.

git bash can be used on windows. 
The idea was to enable first version and then let developers improve them. So I suggest to add README.txt (it should contain information about configuration parameters passed by BB to it as environment variables), create check.sh instead of multiple scripts and  submit them. 

Them switch BB to use these scripts instead of hardcoded configuration.

**Pull request closed without merging**